shim/src/package_manager.rs

```diff
@@ -0,0 +1,85 @@
+use crate::paths::AbsolutePath;
+use anyhow::{anyhow, Result};
+use serde::Deserialize;
+use std::fs;
+
+#[derive(Debug, Deserialize)]
+struct PnpmWorkspaces {
+    pub packages: Vec<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct PackageJsonWorkspaces {
+    pub workspaces: Vec<String>,
+}
+
+pub enum PackageManager {
+    #[allow(dead_code)]
+    Berry,
+    Npm,
+    Pnpm,
+    #[allow(dead_code)]
+    Pnpm6,
+    #[allow(dead_code)]
+    Yarn,
+}
+
+impl PackageManager {
+    /// Returns a list of globs for the package workspace.
+    /// NOTE: We return a `Vec<PathBuf>` instead of a `GlobSet` because we
+    /// may need to iterate through these globs and a `GlobSet` doesn't allow that.
+    ///
+    /// # Arguments
+    ///
+    /// * `root_path`:
+    ///
+    /// returns: Result<Vec<PathBuf, Global>, Error>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///
+    /// ```
+    pub fn get_workspace_globs(&self, root_path: &AbsolutePath) -> Result<Vec<String>> {
+        match self {
+            PackageManager::Pnpm | PackageManager::Pnpm6 => {
+                let workspace_yaml = fs::read_to_string(root_path.join("pnpm-workspace.yaml"))?;
+                let workspaces: PnpmWorkspaces = serde_yaml::from_str(&workspace_yaml)?;
+                if workspaces.packages.is_empty() {
+                    Err(anyhow!("pnpm-workspace.yaml: no packages found. Turborepo requires pnpm workspaces and thus packages to be defined in the root pnpm-workspace.yaml"))
```

Comment:
I know this is a stretch and we don't do this anywhere, but it _miiiight_ be a good idea to start putting error strings into a single file so we have an opportunity to localize them some day.